Readability.js

```diff
@@ -197,20 +197,29 @@ Readability.prototype = {
       return pathBase + uri;
     }
 
-    function convertRelativeURIs(tagName, propName) {
-      var elems = articleContent.getElementsByTagName(tagName);
-      this._forEachNode(elems, function(elem) {
-        var relativeURI = elem.getAttribute(propName);
-        if (relativeURI != null)
-          elem.setAttribute(propName, toAbsoluteURI(relativeURI));
-      });
-    }
-
-     // Fix links.
-    convertRelativeURIs.call(this, "a", "href");
+    var links = articleContent.getElementsByTagName("a");
+    this._forEachNode(links, function(link) {
+      var href = link.getAttribute("href");
+      if (href) {
+        // Replace links with javascript: URIs with text content, since
+        // they won't work after scripts have been removed from the page.
+        if (href.indexOf("javascript:") === 0) {
+          var text = new Text();
+          text.textContent = link.textContent;
+          link.parentNode.replaceChild(text, link);
```

Comment:
Oh, and in relation to the jsdom error... I expect that what is going on is that jsdom incorrectly assumes that the second argument to `replaceChild` has to be an element. (you can't `dispatchEvent` on a text node)
This assumption is wrong, and our code is exposing that problem here... Sadly, I don't really know a good way around this besides using `insertBefore` followed by `removeChild`... except that we don't currently implement `insertBefore` in JSDOMParser...